### PR TITLE
add a simple snapshot session

### PR DESCRIPTION
**Pull request by tbg**
based on discussion on our mailing list:
https://groups.google.com/forum/#!topic/cockroach-db/9Dtbe7ozFzE

Run with

make test PKG=./sql TESTS=Logic TESTFLAGS='-v -d \
testdata/snapshot_unrelated_update'

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3716)

### CODE REVIEW

**Comment by petermattis** [comment]
LGTM

File: sql/testdata/snapshot_unrelated_update [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+query IT
+SELECT * FROM t
+----
+1 A
+2 B
+
+user root
+
+statement ok
+BEGIN TRANSACTION ISOLATION LEVEL SNAPSHOT
```
Comment (vivekmenezes) [inline sql/testdata/snapshot_unrelated_update]: Does this actually create another transaction? I thought you would see errTransactionInProgress

Comment (tschottdorf) [inline sql/testdata/snapshot_unrelated_update]: note we're switching users in l. 26, so this is a new client.

Comment (vivekmenezes) [inline sql/testdata/snapshot_unrelated_update]: ah yes. Thanks

**Pull request merged**